parts/kuberneteswindowssetup.ps1

```
@@ -0,0 +1,299 @@
+        c:\k\nssm set Kubeproxy Start SERVICE_AUTO_START
+        c:\k\nssm set Kubeproxy ObjectName LocalSystem
+        c:\k\nssm set Kubeproxy Type SERVICE_WIN32_OWN_PROCESS
+        c:\k\nssm set Kubeproxy AppThrottle 1500
+        c:\k\nssm set Kubeproxy AppStdout C:\k\kubeproxy.log
+        c:\k\nssm set Kubeproxy AppStderr C:\k\kubeproxy.err.log
+        c:\k\nssm set Kubeproxy AppRotateFiles 1
+        c:\k\nssm set Kubeproxy AppRotateOnline 1
+        c:\k\nssm set Kubeproxy AppRotateSeconds 86400
+        c:\k\nssm set Kubeproxy AppRotateBytes 1048576
+        net start Kubeproxy
+    }
+}
+
+try
+{
+    # Set to false for debugging.  This will output the start script to
+    # c:\AzureData\CustomDataSetupScript.log, and then you can RDP 
+    # to the windows machine, and run the script manually to watch
+    # the output.
+    if ($true) {
+        Write-Log "Provisioning $global:DockerServiceName... with IP $MasterIP"
+
+        Write-Log "download kubelet binaries and unzip"
+        Get-KubeBinaries
+
+        Write-Log "Write kube config"
+        Write-KubeConfig
+
+        Write-Log "Create the Pause Container kubletwin/pause"
+        New-InfraContainer
+
+        Write-Log "Get the POD CIDR"
+        $podCIDR = Get-PodCIDR
+
+        Write-Log "write kubelet startfile with pod CIDR of $podCIDR"
+        Write-KubeletStartFile $podCIDR
+
+        Write-Log "setup docker network with pod CIDR of $podCIDR"
+        Set-DockerNetwork $podCIDR
+
+        Write-Log "install the NSSM service"
+        New-NSSMService
+        
+        Write-Log "Setup Complete"
+    }
+    else 
+    {
+        # keep for debugging purposes
+        Write-Log ".\CustomDataSetupScript.ps1 -MasterIP $MasterIP -KubeDnsServiceIp $KubeDnsServiceIp -MasterFQDNPrefix $MasterFQDNPrefix -Location $Location -CACertificate $CACertificate -AgentCertificate $AgentCertificate -AgentKey $AgentKey -AzureHostname $AzureHostname"
+    }
+}
+catch
+{
+    Write-Error $_

```

Do you want this script to pass as successful in this case? Might want to rethrow the exception and/or fail explicitly?